src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/check_health.py

```diff
@@ -0,0 +1,259 @@
+
+        if stderr:
+            _handle_error(DOCKER_PULL_ERROR.append_error_message(stderr.decode()), ignore_errors)
+        else:
+            if output.find(DOCKER_PULL_SUCCEEDED.format(IMAGE, TAG)) != -1 or \
+                output.find(DOCKER_IMAGE_UP_TO_DATE.format(IMAGE, TAG)) != -1:
+                print_pass("Docker pull")
+            else:
+                _handle_error(DOCKER_PULL_ERROR, ignore_errors)
+
+
+# Get current CLI version
+def _get_cli_version():
+
+    acr_cli_version = "2.2.4"
+    print('ACR CLI version: {}'.format(acr_cli_version))
+
+    return 0
+
+
+# Get helm versions
+def _get_helm_version(ignore_errors):
+    from ._errors import HELM_VERSION_ERROR
+
+    # Helm command check
+    helm_command, error = get_helm_command(is_diagnostics_context=True)
+
+    if error:
+        _handle_error(error, ignore_errors)
+        return
+
+    # Helm version check
+    output, stderr = _subprocess_communicate([helm_command, "version"])
+
+    if stderr:
+        _handle_error(HELM_VERSION_ERROR.append_error_message(stderr.decode()), ignore_errors)
+
+    print("Helm version:\n{}".format(output))
+
+
+def _check_health_environment(ignore_errors, skip_confirmation):
+    from azure.cli.core.util import in_cloud_console
+    if in_cloud_console():
+        print_warning("Environment checks are not supported in Azure Cloud Shell.")
+        return
+
+    _get_docker_status_and_version(ignore_errors, skip_confirmation)
+    _get_cli_version()
+    _get_helm_version(ignore_errors)
+
+
+## Checks for the connectivity ##
+
+
+# Check DNS lookup
+def _get_registry_status(login_server, ignore_errors):
+    import socket
+
+    registry_ip = ""
+
+    try:
+        registry_ip = socket.gethostbyname(login_server)
+    except socket.gaierror:
+        pass
+
+    if not registry_ip:
+        from ._errors import CONNECTIVITY_DNS_ERROR
+        _handle_error(CONNECTIVITY_DNS_ERROR.format_error_message(login_server), ignore_errors)
+        return False
+
+    print_pass("DNS lookup to {} at IP {}".format(login_server, registry_ip))
+
+    import requests
+    from azure.cli.core.util import should_disable_connection_verify
+    challenge = requests.get('https://' + login_server + '/v2/', verify=(not should_disable_connection_verify()))
+
+    if challenge.status_code == 403:
+        from ._errors import CONNECTIVITY_FORBIDDEN_ERROR
+        _handle_error(CONNECTIVITY_FORBIDDEN_ERROR.format_error_message(login_server), ignore_errors)
+        return False
+    return True
+
+
+def _get_endpoint_and_token_status(cmd, login_server, ignore_errors):
+    from ._errors import CONNECTIVITY_CHALLENGE_ERROR, CONNECTIVITY_AAD_LOGIN_ERROR, \
+        CONNECTIVITY_REFRESH_TOKEN_ERROR, CONNECTIVITY_ACCESS_TOKEN_ERROR
+
+    # Check access to login endpoint
+    url = 'https://' + login_server + '/v2/'
+    result_from_token = _get_aad_token(cmd.cli_ctx, login_server, False, is_diagnostics_context=True)
```

Comment:
Can you check if we can refactor and use this method https://github.com/Azure/azure-cli/blob/6868f1f8d8e4b30391426908dcca75f7188d2e7e/src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py#L134 since there is some other checks such as vnet, admin-enabled, etc. so you can also avoid handling suffix here.

Reply:
I was thinking of refactoring this method later, maybe even created a method just to check the dns lookup to the registry and the VNET status and incorporate this in both methods (this one and check health method). The reason for this is because this method as is more specific for logging in. Also we don't want to include checks of admin status on diagnosis.